src/Interpreters/InterpreterAlterQuery.cpp

```diff
@@ -55,7 +55,8 @@ BlockIO InterpreterAlterQuery::execute()
 
     DatabasePtr database = DatabaseCatalog::instance().getDatabase(table_id.database_name);
     if (typeid_cast<DatabaseReplicated *>(database.get())
-        && !getContext()->getClientInfo().is_replicated_database_internal)
+        && !getContext()->getClientInfo().is_replicated_database_internal
+        && !alter.isAttachAlter())
```

Comment:
Btw, should we do the same for [ALTER FETCH PARTITION](https://github.com/ClickHouse/ClickHouse/blob/2dc5c89b66cff541bb66ee16203b87508e870572/src/Interpreters/executeDDLQueryOnCluster.cpp#L42)?